kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/StatsSchemaHelper.java

```diff
@@ -0,0 +1,258 @@
+package io.delta.kernel.internal.skipping;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.internal.util.Tuple2;
+import io.delta.kernel.types.*;
+import static io.delta.kernel.internal.util.ColumnMapping.getPhysicalName;
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+/**
+ * Provides information and utilities for statistics columns given a table schema. Specifically,
+ * it is used to:
+ * <ol>
+ *     <li>Get the expected statistics schema given a table schema</li>
+ *     <li>Check if a {@link Literal} or {@link Column} is skipping eligible</li>
+ *     <li>Get the statistics column for a given stat type and logical column</li>
+ * </ol>
+ */
+public class StatsSchemaHelper {
+
+    //////////////////////////////////////////////////////////////////////////////////
+    // Private static fields and methods
+    //////////////////////////////////////////////////////////////////////////////////
+
+    private static String NUM_RECORDS = "numRecords";
+    private static String MIN = "minValues";
+    private static String MAX = "maxValues";
+    private static String NULL_COUNT = "nullCount";
+
+    private static Set<String> skippingEligibleTypeNames = new HashSet<String>() {
+        {
+            add("byte");
+            add("short");
+            add("integer");
+            add("long");
+            add("float");
+            add("double");
+            add("date");
+            add("timestamp");
+            add("string");
+        }
+    };
+
+    /**
+     * Returns true if the given data type is eligible for MIN/MAX data skipping.
+     */
+    private static boolean isSkippingEligibleDataType(DataType dataType) {
+        return skippingEligibleTypeNames.contains(dataType.toString()) ||
+            dataType instanceof DecimalType;
```

Comment:
why is decimal type a special case?

Reply:
it has a precision and scale :P

Reply:
makes sense. this isn't obvious to a reader. this should have a comment